src/pages/index.js

```diff
@@ -60,66 +97,146 @@ const editCardPopup = new PopupWithForm(
 );
 editCardPopup.setEventListeners();
 
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarEditSubmit
+);
+editAvatarPopup.setEventListeners();
+
+const deleteCardPopup = new PopupWithForm("#delete-card-modal", () => {});
+deleteCardPopup.setEventListeners();
+
 // Import from PopupWithImage
 
 const imagePopUp = new PopupWithImage("#preview-image-modal");
 imagePopUp.setEventListeners();
 
-// Profile DOM nodes
-const profileTitle = document.querySelector(".profile__title");
-const profileDescription = document.querySelector(".profile__description");
-
 // Import from UserInfo
 
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
 // Buttons and other DOM nodes
 const profileEditButton = document.querySelector(".profile__edit-button");
 const addNewCardButton = document.querySelector(".profile__add-button");
-const profileEditCloseButton = profileEditModal.querySelector(".modal__close");
-const addCardModalCloseButton = addCardModal.querySelector(".modal__close");
-const previewImageCloseButton =
-  previewImageModal.querySelector(".modal__close");
+const editAvatarButton = document.querySelector(".profile__image");
+const editAvatarButtonPencil = document.querySelector(".profile__pencil");
 
 // FORM DATA
 const profileTitleInput = profileEditForm.querySelector(".modal__input_name");
 const profileDescriptionInput = profileEditForm.querySelector(
   ".modal__input_description"
 );
-const cardTitleInput = addCardFormElement.querySelector(".modal__input_title");
-const cardLinkInput = addCardFormElement.querySelector(".modal__input_link");
 
 // FUNCTIONS
 
+function handleAvatarEditSubmit(e, formValues) {
+  e.preventDefault();
+
+  const { avatar } = formValues;
+  console.log("Avatar URL being sent:", avatar);
+
+  editAvatarPopup.setLoadingState(false);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```